src/arbitrary.rs

```diff
@@ -0,0 +1,332 @@
+use core::{fmt::Debug, ops::ControlFlow};
+
+use arbitrary::{Arbitrary, Error, Result, Unstructured};
+use cosey::EcdhEsHkdf256PublicKey;
+use heapless::{String, Vec};
+use heapless_bytes::Bytes;
+use serde_bytes::ByteArray;
+
+use crate::{ctap1, ctap2, webauthn};
+
+// cannot be derived because of missing impl for &[T; N]
+impl<'a> Arbitrary<'a> for ctap1::authenticate::Request<'a> {
+    fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
+        let control_byte = Arbitrary::arbitrary(u)?;
+        let challenge = u.bytes(32)?.try_into().unwrap();
+        let app_id = u.bytes(32)?.try_into().unwrap();
+        let key_handle = Arbitrary::arbitrary(u)?;
+        Ok(Self {
+            control_byte,
+            challenge,
+            app_id,
+            key_handle,
+        })
+    }
+}
```

Comment:
nit: So that the fuzzing stays efficient it might be worth implementing the `size_hint` method.

Reply:
→ https://github.com/trussed-dev/ctap-types/issues/58